average.js

```diff
@@ -1,6 +1,16 @@
 
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  // Filtering array to remove NaN
+  const NanRemoved = numbers.filter((num) => !Number.isNaN(num));
+
+  // Calculating number of NaNs and returning NaN if percentage of NaNs is >50
+  const NoOfNaNs = numbers.length -NanRemoved.length;
+  if (NoOfNaNs/numbers.length>0.5) {
```

Comment:
Any reason for choosing 0.5 as the threshold and not another number?

Reply:
Sir, I did some research on this topic but there was no distinct percentage I could find for considering the sensor to be defective. Most articles (like [this](https://stackoverflow.com/questions/51507598/how-do-i-know-whether-to-remove-the-column-or-rows-when-dealing-with-null-data)) decided based on the data type, importance of data and if the NaN values can themselves be computed. So I went by intuition, if more than half the inputs received were NaN then the sensor is probably not acting like it should.